pkgs/development/python-modules/pycairo/default.nix

```diff
@@ -1,50 +1,27 @@
-{ lib, fetchurl, fetchpatch, python, buildPythonPackage, pkgconfig, cairo, xlibsWrapper, isPyPy, isPy35, isPy36, isPy3k }:
+{ lib, fetchFromGitHub, python, buildPythonPackage, pytest, pkgconfig, cairo, xlibsWrapper, isPyPy }:
 
-if (isPyPy) then throw "pycairo not supported for interpreter ${python.executable}" else let
-
-  patch_waf = fetchpatch {
-    url = http://www.linuxfromscratch.org/patches/blfs/8.0/pycairo-1.10.0-waf_python_3_4-1.patch;
-    sha256 = "0xfl1i9dips2nykyg91f5h5r3xpk2hp1js1gq5z0hwjr0in55id4";
-  };
-  patch_waf-py3_5 = ./waf-py3_5.patch;
-
-in buildPythonPackage rec {
+buildPythonPackage rec {
   pname = "pycairo";
-  version = "1.10.0";
+  version = "1.15.4";
   name = "${pname}-${version}";
-  format = "other";
 
-  src = if isPy3k
-    then fetchurl {
-      url = "http://cairographics.org/releases/pycairo-${version}.tar.bz2";
-      sha256 = "1gjkf8x6hyx1skq3hhwcbvwifxvrf9qxis5vx8x5igmmgs70g94s";
-    }
-    else fetchurl {
-      url = "http://cairographics.org/releases/py2cairo-${version}.tar.bz2";
-      sha256 = "0cblk919wh6w0pgb45zf48xwxykfif16qk264yga7h9fdkq3j16k";
-    };
+  disabled = isPyPy;
 
-  patches = [
-    (fetchpatch {
-      url = http://www.linuxfromscratch.org/patches/blfs/8.0/pycairo-1.10.0-waf_unpack-1.patch;
-      sha256 = "1bmrhq2nmhx4l5glvyi59r0hc7w5m56kz41frx7v3dcp8f91p7xd";
-    })
-  ];
+  src = fetchFromGitHub {
+    owner = "pygobject";
+    repo = "pycairo";
+    rev = "v${version}";
+    sha256 = "02vzmfxx8nl6dbwzc911wcj7hqspgqz6v9xmq6579vwfla0vaglv";
+  };
+
+  postPatch = ''
+    # prefix is ${python} by default for some reason
+    substituteInPlace setup.py --replace '"prefix": self.install_base' "'prefix': '$out'"
+  '';
```

Comment:
maybe the following works
```
setupPyBuildFlags = [ "--install-base=$out" ];
```

Reply:
https://stackoverflow.com/questions/5351086/what-exactly-does-the-install-base-parameter-to-setup-py-install-mean

Reply:
Does not seem to work – `no such option: --install-base`. I also tried

    installFlags = [ "--install-base=$out" ];

with the same result, and

    installFlags = [ "--install-option=\"--install-base=$out\"" ];

which at least builds but the `${python}` path is still used.

Reply:
The option seems to be distutils-specific, while this package seems to be using setuptools.

Reply:
Apparently, the problem is caused by pycairo reading an install flag during the build of the wheel. Just like with [passing build flags to bdist_wheel](https://github.com/NixOS/nixpkgs/blob/c0c503ca685f5a70390c8f2bb850e8b9690fab17/pkgs/development/interpreters/python/build-python-package-setuptools.nix#L9), setuptools do not allow passing the install flags either – other than by using hacks like `python setup.py install --root=_temp --prefix=/foo bdist_wheel` ([source](https://quodlibet.duckdns.org/irc/pygobject/python_2017-11-27.log.html#t18:55:09)). Let’s keep patching the source for now.